etcd-backup-operator/pkg/operator/s3backup.go

```diff
@@ -0,0 +1,213 @@
+package operator
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"log"
+	"path"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	apis "github.com/coreos/etcd-operator/pkg/apis/etcd/v1beta2"
+	"github.com/coreos/etcd-operator/pkg/util/constants"
+	"github.com/coreos/etcd-operator/pkg/util/etcdutil"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+	"github.com/coreos/etcd/clientv3"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func (b *Backup) handleS3(clusterName string, s3 *apis.S3Source) error {
+	s3b := &s3Backup{
+		namespace:   b.namespace,
+		name:        b.name,
+		clusterName: clusterName,
+		s3bucket:    s3.S3Bucket,
+		prefix:      ToS3Prefix(s3.Prefix, b.namespace, clusterName),
+		awsSecret:   s3.AWSSecret,
+		kubecli:     b.kubecli,
+	}
+	return s3b.saveSnap()
```

Comment:
It seems unnecessary to define a method for it.
We aren't returning *s3Backup anywhere.